### PR TITLE
[Docs] psm documentation

### PR DESCRIPTION
**Pull request by luarss**
Fixes missing parameters and formatting issues.

This figure is readjusted into table format for easy viewing:
image

https://openroad--3703.org.readthedocs.build/en/3703/main/src/psm/README.html

### CODE REVIEW

File: src/psm/README.md [hunk 1]
```diff
@@ -10,72 +11,138 @@ Features:
...
+Example scripts demonstrating how to run PDNSim on a sample design on `aes` as follows:
+
+```tcl
+./test/aes_test_vdd.tcl
+./test/aes_test_vss.tcl
+```
 
 ## Regression tests
 
+There are a set of regression tests in `/test`. For more information, refer to this [section](../../README.md#regression-tests).
```
Comment (vvbandeira) [inline src/psm/README.md:127]:
```suggestion
There are a set of regression tests in `./test`. For more information, refer to this [section](../../README.md#regression-tests).
```

File: src/psm/README.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
-# PDNSim
+# IR Drop Analysis
 
-PDNSim is an open-source static IR analyzer.
+The IR Drop Analysis module in OpenROAD (`pdn`) is based on PDNSim,
```
Comment (vvbandeira) [inline src/psm/README.md:3]:
```suggestion
The IR Drop Analysis module in OpenROAD (`psm`) is based on PDNSim, 
```

File: src/psm/README.md [hunk 3]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-voltage` | Sets the voltage on a specific net. If this command is not run, the voltage value is obtained from operating conditions in the Liberty file. |
+
+### Useful developer functions
+
+If you are a developer, you might find these useful. More details can be found in the [source file](./src/pdnsim.cpp) or the [swig file](./src/pdnsim.i).
+
+```tcl
+# GUI debugging 
+debug_gui
+```
```
Comment (vvbandeira) [inline src/psm/README.md:114]: Where is this command?

File: src/psm/README.md [hunk 4]
```diff
@@ -10,72 +11,138 @@ Features:
...
-write_pg_spice -vsrc <voltage_source_location_file> -outfile <netlist.sp> -net <net_name>
+
+#### Options
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-vsrc` | File to set the location of the power C4 bumps/IO pins. [Vsrc_aes.loc file](test/Vsrc_aes_vdd.loc) for an example with a description specified [here](doc/Vsrc_description.md). |
+| `-dx`,`-dy` | These arguments set the bump pitch to decide the voltage source location in the absence of a vsrc file. Default bump pitch of 140um used in absence of these arguments and vsrc. |
+| `-net` | Name of the net to analyze, power or ground net name. |
+| `-enable_em` | Flag to report current per power grid segment. |
```
Comment (vvbandeira) [inline src/psm/README.md:50]:
```suggestion
| `-enable_em` | Report current per power grid segment. |
```

File: src/psm/README.md [hunk 5]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-outfile` | Filename specified per-instance voltage written into file. |
+| `-em_outfile` | Filename to write out the per segment current values into a file, can be specified only if `enable_em` flag is set. |
```
Comment (vvbandeira) [inline src/psm/README.md:52]:
```suggestion
| `-em_outfile` | Write the per-segment current values into a file. This option is only available if used in combination with `-enable_em`. |
```

File: src/psm/README.md [hunk 6]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-voltage` | Sets the voltage on a specific net. If this command is not run, the voltage value is obtained from operating conditions in the Liberty file. |
+| `-node_density` | This value can be specfied by the user in microns to determine the node density on the standard cell rails. Cannot be used together with `node_density_factor`. |
```
Comment (vvbandeira) [inline src/psm/README.md:54]:
```suggestion
| `-node_density` | Node density (in microns) on the standard cell rails. It cannot be used together with `-node_density_factor`. |
```

File: src/psm/README.md [hunk 7]
```diff
@@ -10,72 +11,138 @@ Features:
...
+    [-dx bump_pitch_x]
+    [-dy bump_pitch_y]
+    [-corner corner]
+```
+
+#### Options
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-vsrc` | File to set the location of the power C4 bumps/IO pins. [Vsrc_aes.loc file](test/Vsrc_aes_vdd.loc) for an example with a description specified [here](doc/Vsrc_description.md). |
```
Comment (vvbandeira) [inline src/psm/README.md:86]:
```suggestion
| `-vsrc` | File to set the location of the power C4 bumps/IO pins. See [Vsrc_aes.loc file](test/Vsrc_aes_vdd.loc) for an example and its [description](doc/Vsrc_description.md). |
```

File: src/psm/README.md [hunk 8]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-dx`,`-dy` | These arguments set the bump pitch to decide the voltage source location in the absence of a vsrc file. Default bump pitch of 140um used in absence of these arguments and vsrc. |
```
Comment (vvbandeira) [inline src/psm/README.md:87]:
```suggestion
| `-dx`,`-dy` | Set the bump pitch to decide the voltage source location in the absence of a `vsrc` file. The default bump pitch is 140um if neither these arguments nor a `vsrc` file are given. |
```

File: src/psm/README.md [hunk 9]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-net` | Name of the net to analyze, power or ground net name. |
```
Comment (vvbandeira) [inline src/psm/README.md:88]:
```suggestion
| `-net` | Name of the net to analyze. Must be a power or ground net name. |
```

File: src/psm/README.md [hunk 10]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-outfile` | Filename specified per-instance voltage written into file. |
```
Comment (vvbandeira) [inline src/psm/README.md:89]:
```suggestion
| `-outfile` | Write per-instance voltage written into the file. |
```

File: src/psm/README.md [hunk 11]
```diff
@@ -10,72 +11,138 @@ Features:
...
+set_pdnsim_net_voltage
+    [-net net_name]
+    [-voltage volt]
+```
+
+#### Options
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-net` | Name of the net to analyze, power or ground net name. |
```
Comment (vvbandeira) [inline src/psm/README.md:104]:
```suggestion
| `-net` | Name of the net to analyze. It must be a power or ground net name. |
```

File: src/psm/README.md [hunk 12]
```diff
@@ -10,72 +11,138 @@ Features:
...
+| `-voltage` | Sets the voltage on a specific net. If this command is not run, the voltage value is obtained from operating conditions in the Liberty file. |
```
Comment (vvbandeira) [inline src/psm/README.md:105]:
```suggestion
| `-voltage` | Sets the voltage on a specific net. If this option is not given, the Liberty file's voltage value is obtained from operating conditions. |
```

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: src/psm/README.md (hunk 3 above)
Comment (luarss) [reply src/psm/README.md]: Seems it has been removed. Reflected in latest commit

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**